builder/frameworks/espidf.py

```
@@ -195,177 +213,202 @@ def _get_installed_standard_pip_packages():
+        if flash_mode_flag not in config_flags:
+            config_flags += flash_mode_flag
+        
+        # ESP32 specific SPIRAM configuration
+        if mcu == "esp32" and "CONFIG_FREERTOS_UNICORE=y" in config_flags:
+            config_flags += "# CONFIG_SPIRAM is not set\n"
+        
+        return config_flags
+
+    def write_sdkconfig_file(idf_config_flags, checksum_source):
+        """Write the final sdkconfig.defaults file with checksum."""
+        sdkconfig_src = join(arduino_libs_mcu, "sdkconfig")
```

Pylint (3.3.7)

[error] Possibly using variable 'arduino_libs_mcu' before assignment

(E0606)